modules/rubiconBidAdapter.js

```diff
@@ -538,6 +539,12 @@ export const spec = {
       if (sharedId) {
         data['eid_sharedid.org'] = `${sharedId.uids[0].id}^${sharedId.uids[0].atype}^${sharedId.uids[0].ext.third}`;
       }
+      const pubProvidedIds = bidRequest.userIdAsEids.filter(eid => (eid.uids[0] && eid.uids[0].ext && eid.uids[0].ext.stype === 'ppuid'));
+      if (pubProvidedIds) {
+        pubProvidedIds.forEach(pubProvidedId => {
+          data[`${pubProvidedId.source}_id`] = pubProvidedId.uids[0].id;
```

Comment:
As noted above, I think it would be safer to accept eid.uids[].ext.stype === 'ppuid' on any index of the uids array.

More importantly, for the fastlane.json interface, the exchange team wants to receive this on the existing `ppuid` field, not `${pubProvidedId.source}_id`.